### PR TITLE
Removes graph and ref_list modules

### PR DESCRIPTION
**Pull request by Boiethios**

### CODE REVIEW

File: src/stack_height.rs [hunk 1]
```diff
@@ -267,18 +267,14 @@ fn instrument_function(ctx: &mut Context, func: &mut Instructions) -> Result<(),
...
-                            cost,
-                        })
-                    } else {
-                        None
-                    }
-                })
+            if let Call(callee) = *instruction {
+                ctx.stack_cost(callee)
+                    .filter(|&cost| cost > 0)
+                    .map(|cost| InstrumentCall {
```
Comment (mpapierski) [inline src/stack_height.rs:273]: You can do filter_map here instead of filter and map

File: src/optimizer.rs [hunk 2]
```diff
@@ -29,23 +29,20 @@ pub fn optimize(
...
         .entries()
         .iter()
         .enumerate()
-    {
-        if used_exports.iter().any(|e| *e == entry.field()) {
-            stay.insert(Symbol::Export(index));
-        }
-    }
+        .filter(|(_, entry)| used_exports.iter().any(|e| *e == entry.field()))
+        .map(|(index, _)| Symbol::Export(index))
```
Comment (mpapierski) [inline src/optimizer.rs:39]: You can do filter_map here as well

**Review by mpapierski — APPROVED** [review]
LGTM minus nitpick regarding filter_map

File: src/optimizer.rs [hunk 3]
```diff
@@ -29,23 +29,20 @@ pub fn optimize(
...
+        .collect();
 
     // If there is start function in module, it should stary
```
Comment (rafal-ch) [inline src/optimizer.rs:42]:
```suggestion
    // If there is start function in module, it should start
```

File: src/optimizer.rs [hunk 4]
```diff
@@ -99,27 +95,25 @@ pub fn optimize(
...
+        } else {
+            type_section(module)
 					.expect("If type section does not exists, the loop will break at the beginning of first iteration")
 					.types_mut().remove(index);
-                eliminated_types.push(old_index);
-                trace!("Eliminated type({})", old_index);
-            }
-            old_index += 1;
+            eliminated_types.push(old_index);
+            trace!("Eliminated type({})", old_index);
```
Comment (rafal-ch) [inline src/optimizer.rs:114]: The `else` branch could use some formatting, maybe use the `cargo fmt` on the entire code.

**Review by rafal-ch — APPROVED** [review]
No blockers, LGTM :+1:

File: src/stack_height.rs (hunk 1 above)
Comment (mpapierski) [reply src/stack_height.rs:273]: Nevermind, I forgot it's an Option

File: src/optimizer.rs (hunk 4 above)
Comment (mpapierski) [reply src/optimizer.rs:114]: Yeah, original authors used tabs, but we tried to normalize it to indent with spaces.

File: src/optimizer.rs (hunk 3 above)
Comment (mpapierski) [reply src/optimizer.rs:42]: Fixed

**Pull request merged**